src/install/index.js

```diff
@@ -0,0 +1,53 @@
+const core = require('@actions/core');
+const toolCache = require('@actions/tool-cache');
+const cache = require('@actions/cache');
+const exec = require('@actions/exec');
+
+const TP_INSTALL_CACHE_ID = 'telepresence-install-id';
+
+const windowsInstall = async (version) => {
+    core.setFailed('Not implemented for use with Windows runners');
+};
+
+const unixInstall = async  (version) => {
+    const cacheKey = TP_INSTALL_CACHE_ID + `-${version}`;
+    const TELEPRESENCE_DOWNLOAD_URL = process.platform === 'darwin' ?
+        `https://app.getambassador.io/download/tel2/darwin/amd64/${version}/telepresence` :
+        `https://app.getambassador.io/download/tel2/linux/amd64/${version}/telepresence`;
+
+    let tpCacheId = await cache.restoreCache([this.TP_PATH], cacheKey);
+
+    if (!tpCacheId) {
+        try {
+            await toolCache.downloadTool(TELEPRESENCE_DOWNLOAD_URL, `${this.TP_PATH}/telepresence`);
+            tpCacheId = await cache.saveCache([this.TP_PATH], cacheKey);
+            if(!tpCacheId)
+                core.setFailed('There was a problem saving the telepresence binary.');
```

Comment:
`core.setFailed` doesn't stop the execution of the function. If we want to continue with the execution we'd probably want a `warning` instead